### PR TITLE
Add CreateArray and CreateMap to Helper

### LINKED ISSUE #2143: How to push a ContractParameterType of type 'Map' using the script builder?
Hi, 

Some of our code has broken because we can't push the Map type using the script builder. What should we do to replace it? Map is still a ContractParameterType, but it can't be pushed to a script using the script builder.

Edit: Maybe this is not the reason our code has broken. I checked the latest changes and I noticed that the map was never there. Not sure what happened.

### PR DESCRIPTION
**Pull request by erikzhang**
Close #2143

### CODE REVIEW

**Comment by erikzhang** [comment]
@lock9

**Comment by lock9** [comment]
CODING MACHINE
Thanks man, let me check

**Comment by lock9** [comment]
image

The code looks good, but I think some UT is recommended. Thanks a lot for the super speed Erik!

File: src/neo/VM/Helper.cs [hunk 1]
```diff
@@ -15,6 +15,30 @@ namespace Neo.VM
...
+        {
+            if (list is null || list.Count == 0)
+                return sb.Emit(OpCode.NEWARRAY0);
+            for (int i = list.Count - 1; i >= 0; i--)
+                sb.EmitPush(list[i]);
+            sb.EmitPush(list.Count);
+            return sb.Emit(OpCode.PACK);
+        }
+
+        public static ScriptBuilder CreateMap<TKey, TValue>(this ScriptBuilder sb, IReadOnlyDictionary<TKey, TValue> map = null)
```
Comment (lock9) [inline src/neo/VM/Helper.cs:28]: Shouldn't we keep the naming "Emit" or "Push"?

Comment (erikzhang) [reply src/neo/VM/Helper.cs:28]: Because it does not directly push the map, it actually creates a new map.

**Review by shargon — APPROVED** [review]

**Comment by lock9** [comment]
Hello @shargon and @erikzhang , if we don't add these instructions to the Emit method, we create a situation where you can use the Emit method, but it will fail with an exception. In a few words: it compiles but fails in runtime:

image

**Comment by shargon** [comment]
@lock9 but `ContractParameter` is not a `IReadOnlyDictionary` or `IReadOnlyList`. Could you provide an example?

**Pull request merged**